server/src/middleware/validate_token.rs

```diff
@@ -19,76 +20,227 @@ pub async fn validate_token(
         .unwrap()
         .clone()
         .into_inner();
+
+    let validation_status = validate(&token, maybe_validator, &token_cache, req.path()).await;
```

Comment:
All of the logic here is delegated out to the private, pure validate function and only so we can test the thing without hundreds of lines of boilerplate

Reply:
Love it